core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -144,7 +148,7 @@ public void run() {
                         // those (and they randomize the client node usage, so tricky to find when)
                         onFailure(idx, nodeId, new NodeShouldNotConnectException(clusterService.localNode(), node));
                     } else {
-                        NodeRequest nodeRequest = newNodeRequest(nodeId, request);
```

Comment:
any chance we can make the `parentTaskNode` and the `parentTaskId` lazy and just expose them as package private setters. That way we don't need to pass them to all nodeRequests but only set them on the node requests that are tasks requests. This is really polluting the interface IMO.